Mimir/Repositories/StakeRepository.cs

```diff
@@ -1,26 +1,29 @@
 using Libplanet.Crypto;
 using Mimir.Exceptions;
 using Mimir.MongoDB;
+using Mimir.MongoDB.Bson;
 using Mimir.Services;
 using MongoDB.Driver;
 
 namespace Mimir.Repositories;
 
 public class StakeRepository(MongoDbService dbService)
 {
-    // public async Task<StakeDocument> GetByAddressAsync(Address agentAddress)
-    // {
-    //     var collectionName = CollectionNames.GetCollectionName<StakeDocument>();
-    //     var collection = dbService.GetCollection<StakeDocument>(collectionName);
-    //     var filter = Builders<StakeDocument>.Filter.Eq("Address", agentAddress.ToHex());
-    //     var document = await collection.Find(filter).FirstOrDefaultAsync();
-    //     if (document is null)
-    //     {
-    //         throw new DocumentNotFoundInMongoCollectionException(
-    //             collection.CollectionNamespace.CollectionName,
-    //             $"'Address' equals to '{agentAddress.ToHex()}'");
-    //     }
-    //
-    //     return document;
-    // }
+    public async Task<StakeDocument> GetByAgentAddressAsync(Address agentAddress)
+    {
+        var collectionName = CollectionNames.GetCollectionName<StakeDocument>();
+        var collection = dbService.GetCollection<StakeDocument>(collectionName);
+        var filter = Builders<StakeDocument>.Filter.Eq("AgentAddress", agentAddress.ToHex());
+        var doc = await collection.Find(filter).FirstOrDefaultAsync();
+
+        if (doc is null)
+        {
+            throw new DocumentNotFoundInMongoCollectionException(
+                collection.CollectionNamespace.CollectionName,
+                $"'Address' equals to '{agentAddress.ToHex()}'"
```

Comment:
I'm not sure about the usage of the `DocumentNotFoundInMongoCollectionException` class, it should be like the below suggestion when I looked the `filter` definition.

```suggestion
                $"'AgentAddress' equals to '{agentAddress.ToHex()}'"
```

Reply:
I believe it's being used in `HttpResponseFormatter`

Reply:
And thanks you